.githooks/pre-commit

```diff
@@ -0,0 +1,26 @@
+#! /usr/bin/env sh
+#
+# This hook runs on `git commit` and will prevent you from committing without
+# approval from the linter and tests.
+#
+# To run, this file must be symlinked to:
+# .git/hooks/pre-commit
+#
+# To bypass this hook, run:
+# $ git commit --no-verify
+# $ git commit -n
+
+echo "Running linter..."
+bundle exec rake standard
+linter_status=$?
+
+echo "Running tests..."
+bundle exec rake spec
+test_status=$?
+
+if [ "${linter_status}" = 0 ] && [ "${test_status}" = 0 ]; then
+    exit 0
+else
+    echo "Roll your lint and fix your tests before committing!"
```

Comment:
[question] Hmm, running the tests before every commit will probably get annoying when we have a lot of tests. would it make more sense to run the linters and just let CI deal with tests? (i guess since it's a symlinked file everyone can decide for themselves)

Reply:
That's true! That works for me if you all think it's a good idea.

Reply:
i think that's a good idea. we could have a pre-push githook if you think it's important to run tests locally before pushing! (although realistically, this github action that i am trying to set up should be guard enough for tests.)